xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -251,31 +221,48 @@ String getType() {
 
     @Nullable
     Map<String, ?> getConfig() {
-      return config;
+      if (config != null) {
+        return Collections.unmodifiableMap(config);
```

Comment:
Why change the existing code? Even change, why `unmodifiableMap` here, not in the constructor? You return a new map each time you call `getConfig()`

Reply:
This class is immutable, `config` should only represents the internal data representation. If we present this to the user, we should not leak this. It only returns an unmodifiable view of the specified map, isn't as expensive as creating a new one.